### PR TITLE
Update bear to v3.0.0

### PR DESCRIPTION
**Pull request by vmiheer**

### CODE REVIEW

**Comment by vmiheer** [comment]
@trws, will you please review? Also would you like to be primary maintainers of this package?

**Review by trws — CHANGES_REQUESTED** [review]
I would not, I haven't been using it for quite a while.  That said, this looks mostly right, the cmake dependency needs to be upped to 3.12 for version 3 and up though.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please add a `version` directive for 3.0.0.  The sha256 will be `7b68aad69e887d945ad20f8e9f3a8c33cf2d59cc80da7e52d931d8c685fe2f79`.

**Comment by tldahlgren** [comment]
Accidentally clicked the wrong button, have re-opened.

File: var/spack/repos/builtin/packages/bear/package.py [hunk 1]
```diff
@@ -15,5 +15,9 @@ class Bear(CMakePackage):
     version('2.2.0', sha256='6bd61a6d64a24a61eab17e7f2950e688820c72635e1cf7ea8ea7bf9482f3b612')
     version('2.0.4', sha256='33ea117b09068aa2cd59c0f0f7535ad82c5ee473133779f1cc20f6f99793a63e')
```
Comment (trws) [inline var/spack/repos/builtin/packages/bear/package.py:21]:
```suggestion
    version('2.0.4', sha256='33ea117b09068aa2cd59c0f0f7535ad82c5ee473133779f1cc20f6f99793a63e')
    version('3.0.0', sha256='7b68aad69e887d945ad20f8e9f3a8c33cf2d59cc80da7e52d931d8c685fe2f79')
```
(edit) as tammy points out, this is placed wrong, resolving this one and leaving the corrected one above.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/bear/package.py:21]: The convention is to have newer versions appear *before* older ones, so the directive should be added *before* `2.0.4`.

File: var/spack/repos/builtin/packages/bear/package.py [hunk 2]
```diff
@@ -15,5 +15,9 @@ class Bear(CMakePackage):
     version('2.2.0', sha256='6bd61a6d64a24a61eab17e7f2950e688820c72635e1cf7ea8ea7bf9482f3b612')
```
Comment (trws) [inline var/spack/repos/builtin/packages/bear/package.py:20]:
```suggestion
    version('3.0.0', sha256='7b68aad69e887d945ad20f8e9f3a8c33cf2d59cc80da7e52d931d8c685fe2f79')
    version('2.2.0', sha256='6bd61a6d64a24a61eab17e7f2950e688820c72635e1cf7ea8ea7bf9482f3b612')
```

**Comment by vmiheer** [comment]
I have updated commit with all issues addressed.

**Review by trws — APPROVED** [review]

**Comment by vmiheer** [comment]
This error in test seems unrelated: https://github.com/spack/spack/runs/6619693880?check_suite_focus=true#step:6:1301

**Review by tldahlgren — APPROVED** [review]
Confirmed the new sha256.

**Pull request merged**